rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -240,11 +238,13 @@ NodeBase::get_default_callback_group()
 bool
 NodeBase::callback_group_in_node(rclcpp::CallbackGroup::SharedPtr group)
 {
+  std::lock_guard<std::mutex> lock(callback_groups_mutex_);
   bool group_belongs_to_this_node = false;
   for (auto & weak_group : this->callback_groups_) {
     auto cur_group = weak_group.lock();
     if (cur_group && (cur_group == group)) {
       group_belongs_to_this_node = true;
+      break;
```

Comment:
```suggestion
      return true;
```

Reply:
I've also fixed this up in the latest code.